src/backend/src/services/teams.services.ts

```diff
@@ -34,7 +34,33 @@ export default class TeamsService {
 
     return teamTransformer(team);
   }
+  /**
+   * Changes the description of the given team to be the new description
+   * @param user The user who is editing the description
+   * @param teamId The id for the team that is being edited
+   * @param newDescription the new description for the team
+   * @returns The team with the new description
+   */
+  static async editDescription(user: User, teamId: string, newDescription: string): Promise<Team> {
+    const team = await prisma.team.findUnique({
+      where: { teamId },
+      ...teamQueryArgs
+    });
+
+    if (!team) throw new NotFoundException('Team', teamId);
+    if (user.role === Role.GUEST || (user.role === Role.MEMBER && user.userId !== team.leaderId))
```

Comment:
there's also the leadership role... i think this should be if (!(admin or app admin or team lead))